### PR TITLE
Implement initial bot

### PR DESCRIPTION
**Pull request by theArgo15**
Bot can create a channel within a category.

### CODE REVIEW

File: bullyBot.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        f'{guild.name}(id: {guild.id})'
+    )
+
+@bot.command(name='bully', help='Creates a new channel for bullying a new member')
+async def on_message(ctx, victim):
+    
+
+
+    victim=victim.lower().strip()
+    channel_name=f'making-fun-of-{victim}'
```
Comment (EgMan) [inline bullyBot.py]: run linter on this file

**Comment by EgMan** [comment]
Executables shouldn't be in this PR either.  But that begs the question: how did the python executables get into this PR?  Did you make this project in a sub-directory of your python installation, then you initialized the repo in your python installation?

File: bullyBot.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        if guild.name == GUILD:
+            break
+
+    print(
+        f'{bot.user.name} is connected to the following guild:\n'
+        f'{guild.name}(id: {guild.id})'
+    )
+
+@bot.command(name='bully', help='Creates a new channel for bullying a new member')
+async def on_message(ctx, victim):
```
Comment (EgMan) [inline bullyBot.py]: What happens when a message comes in with the wrong number of arguments? We should display a help dialog.

File: bullyBot.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    victim=victim.lower().strip()
+    channel_name=f'making-fun-of-{victim}'
+    guild = ctx.guild
+    existing_channel = discord.utils.get(guild.channels, name=channel_name)
+    bullying_category = discord.utils.get(guild.categories, name='bullying')
+
+    if not existing_channel:
+        response = f'Good choice, {victim} is due for a bullying'
+        #print(f'Creating a new channel for the purpose of bullying {victim}.')
```
Comment (EgMan) [inline bullyBot.py]: General rule: unprofessional to commit commented out code.  Decided whether or not we want this debug message.

**Comment by theArgo15** [comment]
Oh boy I am going to have a hard time helping you understand via text.
I used venv to create a virtual environment for this project. This is my
first time using a virtual environment. I believe that it makes its own
python installation. You were with me when I did the git init, so I don't
know where exactly I was in the file structure.

File: bullyBot.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+#bullyBot.py
```
Comment (EgMan) [inline bullyBot.py]: I think this was supposed to be a shebang.  Shebangs help command lines know what interpreter to use without having to be explicit.  Google python shebang.

File: bullyBot.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    existing_channel = discord.utils.get(guild.channels, name=channel_name)
+    bullying_category = discord.utils.get(guild.categories, name='bullying')
+
+    if not existing_channel:
+        response = f'Good choice, {victim} is due for a bullying'
+        #print(f'Creating a new channel for the purpose of bullying {victim}.')
+        await guild.create_text_channel(channel_name, category=bullying_category)
+
+    else:
+        response = f'Hey now, we already have a channel for bullying {victim}'
```
Comment (EgMan) [inline bullyBot.py]: TODO: lets pull each response from an an array of possible responses, giving the bot more than one way to respond to respond to each event, which gives it a bit more personality.  +1

File: bullyBot.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    
+
+
+    victim=victim.lower().strip()
+    channel_name=f'making-fun-of-{victim}'
+    guild = ctx.guild
+    existing_channel = discord.utils.get(guild.channels, name=channel_name)
+    bullying_category = discord.utils.get(guild.categories, name='bullying')
+
+    if not existing_channel:
```
Comment (EgMan) [inline bullyBot.py]: TODO: swap this out for a fuzzy search (low importance). +1

File: bullyBot.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+#bullyBot.py
+import os
+import discord
+from discord.ext import commands
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (EgMan) [inline bullyBot.py:7]: Why do we need to load dot environment? I don't see the use of any environment variables or anything.  Check if this bot still works without this.

Comment (EgMan) [inline bullyBot.py:9]: Why do we need to load dot environment? I don't see the use of any environment variables or anything. Check if this bot still works without this.

**Comment by EgMan** [comment]
Code looks good for the most part.  Just some nit picky things.  Lets get on a call soon.

File: bullyBot.py (hunk 7 above)
Comment (theArgo15) [reply bullyBot.py:9]: My bot token lives in an environment variable.

File: bullyBot.py (hunk 5 above)
Comment (theArgo15) [reply bullyBot.py]: I like this idea.

File: bullyBot.py (hunk 3 above)
Comment (theArgo15) [reply bullyBot.py]: Thanks for the feedback. I did not know that rule.

File: bullyBot.py (hunk 7 above)
Comment (EgMan) [reply bullyBot.py:9]: Makes sense +1

File: bullyBot.py (hunk 1 above)
Comment (theArgo15) [reply bullyBot.py]: Ran linter

File: bullyBot.py (hunk 2 above)
Comment (theArgo15) [reply bullyBot.py]: Help dialog added.

File: bullyBot.py (hunk 4 above)
Comment (theArgo15) [reply bullyBot.py]: Added shebang

File: bullyBot.py (hunk 5 above)
Comment (theArgo15) [reply bullyBot.py]: Added several responses

File: bullyBot.py (hunk 6 above)
Comment (theArgo15) [reply bullyBot.py]: Added todo comment in code

File: bullyBot.py (hunk 1 above)
Comment (EgMan) [reply bullyBot.py]: +1

File: bullyBot.py (hunk 2 above)
Comment (EgMan) [reply bullyBot.py]: +1

File: bullyBot.py (hunk 3 above)
Comment (EgMan) [reply bullyBot.py]: +1

File: bullyBot.py (hunk 4 above)
Comment (EgMan) [reply bullyBot.py]: +1

File: bullyBot.py (hunk 5 above)
Comment (EgMan) [reply bullyBot.py]: niiiice +1

File: bullyBot.py (hunk 6 above)
Comment (EgMan) [reply bullyBot.py]: Usually TODO comments literally have "TODO" in their text.  That way you can do a text search for them to find all of them.  
+1

**Review by EgMan — APPROVED** [review]

**Pull request merged**